From 928a660ec1124853d2dae074e74ec7b20fe9bac2 Mon Sep 17 00:00:00 2001
From: Zoltan HERPAI <wigyori@uid0.hu>
Date: Sun, 1 Jun 2025 16:02:38 +0000
Subject: [PATCH] riscv: dts: starfive: visionfive2: add dma pool entry

In the VF2 SDK there is a reserved memory for a shared dma pool, which is
also updated by the SDK bootloader. Add this node here as well.

Signed-off-by: Zoltan HERPAI <wigyori@uid0.hu>
---
 .../boot/dts/starfive/jh7110-starfive-visionfive-2.dtsi  | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/arch/riscv/boot/dts/starfive/jh7110-starfive-visionfive-2.dtsi
+++ b/arch/riscv/boot/dts/starfive/jh7110-starfive-visionfive-2.dtsi
@@ -25,6 +25,15 @@
 		e24_mem: e24@c0000000 {
 			reg = <0x0 0x6ce00000 0x0 0x1600000>;
 		};
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x0 0x20000000>;
+			alignment = <0x0 0x1000>;
+			alloc-ranges = <0x0 0x70000000 0x0 0x20000000>;
+			linux,cma-default;
+		};
 	};
 
 	leds {
